shared/charts/gpii-flushtokens/templates/cronjob.yaml

```diff
@@ -0,0 +1,51 @@
+apiVersion: batch/v1beta1
+kind: CronJob
+metadata:
+  namespace: {{ .Release.Namespace | quote }}
+  name: {{ template "flushtokens.name" . }}
+spec:
+  schedule: "{{ .Values.cronJobSchedule }}"
+  successfulJobsHistoryLimit: 5
+  failedJobsHistoryLimit: 1
+  concurrencyPolicy: Replace
+  startingDeadlineSeconds: 200
+  jobTemplate:
+    spec:
+      template:
+        metadata:
+          labels:
+            app: {{ template "flushtokens.name" . }}
+        spec:
+          shareProcessNamespace: true
+          containers:
+          - name: gpii-flushtokens
+            image: "{{ .Values.image.repository }}@{{ .Values.image.checksum }}"
+            command: [ "/bin/sh", "-c", "/app/scripts/dockerDeleteExpiredAccessTokens.sh" ]
+            env:
+            - name: GPII_COUCHDB_URL
+              value: '{{ .Values.couchdbUrl }}'
+            # Use '--deleteAll' to delete all of the access tokens
+            - name: DELETE_ALL
+              value: 'false'
+          - name: istio-proxy-manager
+            image: "{{ .Values.image.repository }}@{{ .Values.image.checksum }}"
+            securityContext:
+              allowPrivilegeEscalation: false
+              runAsUser: 1337
+            command: [ "/bin/sh", "-c" ]
+            args:
+              - while ! ps -o user | grep node > /dev/null;
+                do
+                  echo 'Waiting for dockerDeleteExpiredAccessTokens.sh to start...';
+                  sleep 2;
+                done;
+                echo 'dockerDeleteExpiredAccessTokens.sh started.';
+                while ps -o user | grep node > /dev/null;
+                do
+                  echo 'Waiting for dockerDeleteExpiredAccessTokens.sh to finish...';
+                  sleep 2;
+                done;
+                echo 'dockerDeleteExpiredAccessTokens.sh finished.';
+                echo 'Sending TERM to pilot-agent';
+                pkill pilot-agent;
```

Comment:
I know you and Stepan tried a bunch of stuff so maybe you already covered this:

Would it be valuable to have a `while` loop similar to the loops above to make sure `pilot-agent` really terminates? If `pkill` fails here for some reason, it looks like the current implementation would suffer from the container-never-dying problem.

Reply:
I thought of that but could not see why pkill would fail, or, if it did fail, then it would always fail.  For example, if the process ignore SIGTERM, then it will always ignore SIGTERM.

If so, there is little difference between the istio-proxy container never completing and both of the istio-proxy and istio-proxy-manager containers never completing.

However, putting in another loop is no big deal, and I'll do it.

Reply:
I agree that pkill is unlikely to fail, but I'm sure it's possible and you have the logic right there.

I considered escalating to SIGKILL if SIGTERM fails but in the end I thought that would be, to coin a phrase, overkill.